lib/client.js

```diff
@@ -153,10 +153,8 @@ SwaggerClient.prototype.initialize = function (url, options) {
     this.scheme = options.scheme;
   }
 
-  if (this.usePromise || typeof options.success === 'function') {
-    this.ready = true;
-    return this.build();
-  }
+  this.ready = true;
```

Comment:
Instead of removing this check, would it be more comfortable to check if options.url or options.spec are set? It doesn't effect the existing test cases either way